document/src/vespa/document/fieldset/fieldsets.cpp

```diff
@@ -4,30 +4,35 @@
 #include <vespa/document/fieldvalue/document.h>
 #include <vespa/document/datatype/documenttype.h>
 #include <vespa/vespalib/stllike/asciistream.h>
+#include <algorithm>
 #include <xxhash.h>
 
 namespace document {
 
 namespace {
 
 uint64_t
-computeHash(const Field::Set & set) {
-    if (set.empty()) return 0ul;
+computeHash(const FieldCollection::FieldList & list) {
+    if (list.empty()) return 0ul;
 
     vespalib::asciistream os;
-    for (const Field * field : set) {
+    for (const Field * field : list) {
```

Comment:
I think xxhash has support for incremental hash generation, which might be faster (will still need some form of field separator, though)